jetty-osgi/jetty-osgi-boot/src/main/java/org/eclipse/jetty/osgi/boot/internal/serverfactory/ServerInstanceWrapper.java

```diff
@@ -470,4 +471,14 @@ private void init()
         return files;
     }
 
+    /**
+     * A Resource suitable for use the Bad/Unpredictable URLs from {@link org.osgi.framework.Bundle#getEntry(String)}
+     */
+    public static class BundleResource extends URLResource
```

Comment:
I'd rather call this something different to avoid confusion, it's really got nothing to do with the bundle protocol.  Also, presumably there's some benefit in using the PathResource for file: urls, otherwise we'd just stick to using the URLResource for that too, so what are we missing out on by using the URLResource instead?

Reply:
I might just get rid of this.